streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/VariableServiceImpl.java

```diff
@@ -17,48 +17,79 @@
 
 package org.apache.streampark.console.core.service.impl;
 
+import org.apache.streampark.common.util.DeflaterUtils;
 import org.apache.streampark.console.base.domain.RestRequest;
 import org.apache.streampark.console.base.exception.ApiAlertException;
 import org.apache.streampark.console.base.mybatis.pager.MybatisPager;
+import org.apache.streampark.console.core.entity.Application;
+import org.apache.streampark.console.core.entity.FlinkSql;
 import org.apache.streampark.console.core.entity.Variable;
 import org.apache.streampark.console.core.mapper.VariableMapper;
 import org.apache.streampark.console.core.service.ApplicationService;
 import org.apache.streampark.console.core.service.CommonService;
+import org.apache.streampark.console.core.service.FlinkSqlService;
 import org.apache.streampark.console.core.service.VariableService;
 
 import com.baomidou.mybatisplus.core.conditions.query.LambdaQueryWrapper;
 import com.baomidou.mybatisplus.core.metadata.IPage;
 import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
 import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 @Slf4j
 @Service
 @Transactional(propagation = Propagation.SUPPORTS, readOnly = true, rollbackFor = Exception.class)
 public class VariableServiceImpl extends ServiceImpl<VariableMapper, Variable> implements VariableService {
 
+    private final Pattern placeholderPattern = Pattern.compile("\\$\\{([A-Za-z])+([A-Za-z0-9._-])+\\}");
```

Comment:
If `placeholderStart` and `placeholderEnd ` can be configured, the placeholderPattern should be dynamic.

And I'm afraid, there may be bugs here. 

Hi @macksonmu @wolfboys , I don't recommend supporting configuration `placeholder.start` and `placeholder.end`. Because:

- If placeholderStart is `a`, and `placeholderEnd` is `b`, it’s very easy to conflict.
- It will also become harder if we want to automatically search for variables when we type $ in the future.

Reply:
> If `placeholderStart` and `placeholderEnd ` can be configured, the placeholderPattern should be dynamic.
> 
> And I'm afraid, there may be bugs here.
> 
> Hi @macksonmu @wolfboys , I don't recommend supporting configuration `placeholder.start` and `placeholder.end`. Because:
> 
> * If placeholderStart is `a`, and `placeholderEnd` is `b`, it’s very easy to conflict.
> * It will also become harder if we want to automatically search for variables when we type $ in the future.


I agree with you, placeholder.start and placeholder.end supporting configurable, It does cause unpredictable problems that are difficult to solve.